build.gradle

```diff
@@ -195,21 +195,81 @@ task javadoc(type:GradleBuild) {
     configure copyProperties
 }
 
-task javadocPackage(type: Zip) {
-    description = 'Generate a Zip file with all SDK docs'
+// Find property in either System environment or Gradle properties.
+// If set in both places, Gradle properties win.
+def getPropertyValueOrThrow(String propertyName) {
+    def value = System.getenv(propertyName)
+    if (project.hasProperty(propertyName)) {
+        value = project.getProperty(propertyName)
+    }
+    if (value == null || value.trim().isEmpty()) {
+        throw new GradleException("Could not find '$propertyName'. " +
+                "Most be provided as either environment variable or " +
+                "a Gradle property.")
+    }
+    return value
+}
+
+task createLatestJavadocRedirectFile(type: Copy) {
+    description = 'Redirects from /java/latest/ to correct version'
+    from 'realm/templates'
+    into "$buildDir/outputs/doc_redirects/java_latest"
+    include 'redirect.html.template'
+    rename { file -> 'index.html' }
+    expand(title: "Realm Java ${currentVersion}", url: "../${currentVersion}/index.html")
+}
+
+task createLatestKotlindocRedirectFile(type: Copy) {
+    description = 'Redirects from /kotlin/latest/ to correct version'
+    from 'realm/templates'
+    into "$buildDir/outputs/doc_redirects/kotlin_latest"
+    include 'redirect.html.template'
+    rename { file -> 'index.html' }
+    expand(title: "Kotlin Extensions ${currentVersion}", url: "../${currentVersion}/index.html")
+}
+
+task createKotlinRootRedirectFile(type: Copy) {
+    description = 'Redirects from /kotlin/<version>/ to /kotlin/<version>/kotlin-extensions, which is the real root folder'
+    from 'realm/templates'
+    into "$buildDir/outputs/doc_redirects/kotlin_root"
+    include 'redirect.html.template'
+    rename { file -> 'index.html' }
+    expand(title: "Kotlin Extensions ${currentVersion}", url: "./kotlin-extensions/index.html")
+}
+
+task uploadJavadoc {
```

Comment:
Is it actually called from somewhere? Or maybe it was just automated yet?

Reply:
No, it isn't called from anywhere automatically yet. Just a manual step you have to call as part of releasing. But I was hoping to be able to automate this soon so a release is just a matter of tagging a commit